### ISSUE TITLE
selector-no-id errors on Sass interpolation `#{ }`

### ISSUE THREAD

**Issue by timothyneiljohnson** [stylelint/stylelint#1005]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where using interpolation in Sass causes the selector-no-id error to be thrown.

> Which rule, if any, is this issue related to?

selector-no-id

> What CSS is needed to reproduce this issue?

```css
@keyframes loader {
  0% {
    transform: translate3d(0, 0, 0);
  }
  #{50% - $loader-icon-duration} { // Throwing error
    transform: translate3d(0, $bounce-height, 0);
  }
  50% {
    transform: translate3d(0, $bounce-height, 0) scale($loader-bounce-horizontal-expansion, $loader-bounce-vertical-compression);
  }
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "selector-no-id": true
  }
}
```

> Which version of stylelint are you using?

e.g. `5.0.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

PostCSS plugin

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS interpolation.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
Unexpected id selector (selector-no-id)
```

**Comment by davidtheclark** [comment]
There seem to be two issues here:

- One is essentially the same as #945: postcss-selector-parser interprets Sass interpolation as ids and that messes with id-related-rules. So that fix is hopefully coming.
- The other is that selector-no-id should not even be paying attention to `@keyframes` selectors. That's an easy fix.
